### ISSUE TITLE
failure in TestTransfer

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#2180]
See https://circleci.com/gh/weaveworks/weave/5402
```
--- FAIL: TestTransfer (0.06s)
	Error Trace:	allocator_test.go:361
	Error:		Not equal: 0x20 (expected)
			        != 0x40 (actual)
```

**Comment by rade** [comment]
With
```patch
diff --git a/ipam/allocator_test.go b/ipam/allocator_test.go
index 5df6060..287e4eb 100644
--- a/ipam/allocator_test.go
+++ b/ipam/allocator_test.go
@@ -369,6 +369,12 @@ func TestTransfer(t *testing.T) {
        alloc1.Stop()
 }
 
+func TestTransfer1000(t *testing.T) {
+       for i := 0; i < 1000; i++ {
+               TestTransfer(t)
+       }
+}
+
 func TestFakeRouterSimple(t *testing.T) {
        const (
                cidr = "10.0.1.7/22"
```
and running

    GO15VENDOREXPERIMENT=1 go test -v -run TestTransfer1000 ./ipam

I have managed to get
```
--- FAIL: TestTransfer1000 (0.27s)
	Error Trace:	allocator_test.go:361
	Error:		Not equal: 0x80 (expected)
			        != 0x100 (actual)	
```
on my desktop.

**Comment by rade** [comment]
I cannot reproduce this on 1.4; the test has changed a fair bit on master. However, the underlying cause is in fact present in 1.4, and the fix is easy, so we might as well put it there.

**Comment by rade** [comment]
better fix in d3635eed77c
